core/lib/slider.ts

```diff
@@ -0,0 +1,509 @@
+	 */
+	readonly: boolean;
+
+	/**
+	 * If `true` slider value cannot be changed and the slider cannot be focused
+	 */
+	disabled: boolean;
+
+	/**
+	 * If `true` is vertically positioned otherwise it is horizontal
+	 */
+	vertical: boolean;
+
+	/**
+	 * Current slider values
+	 */
+	values: number[];
+}
+
+export interface SliderState extends SliderCommonPropsAndState {
+	/**
+	 * Progress bar left position in %
+	 */
+	progressLeft: number[];
+
+	/**
+	 * Progress bar bottom position in %
+	 */
+	progressBottom: number[];
+
+	/**
+	 * Progress bar height in %
+	 */
+	progressHeight: number[];
+
+	/**
+	 * Progress bar width in %
+	 */
+	progressWidth: number[];
+
+	/**
+	 * Sorted slider values
+	 */
+	sortedValues: number[];
+
+	/**
+	 * Combined label left offset in %
+	 */
+	combinedLabelPositionLeft: number;
+
+	/**
+	 * Combined label top offset in %
+	 */
+	combinedLabelPositionTop: number;
+
+	/**
+	 * Array of the left offsets for the tooltips in %
+	 */
+	handleTooltipLeft: number[];
+
+	/**
+	 * Array of the top offsets for the tooltips in %
+	 */
+	handleTooltipTop: number[];
+
+	/**
+	 * If `visible` the minimum label will be displayed
+	 */
+	minValueLabelDisplay: string;
+
+	/**
+	 * If `visible` the maximum label will be displayed
+	 */
+	maxValueLabelDisplay: string;
+
+	/**
+	 * If `visible` the label when the handles are close is displayed
+	 */
+	combinedLabelDisplay: string;
+
+	/**
+	 * Array of the sorted handles to display
+	 */
+	sortedHandles: {value: number; id: number}[];
+}
+
+export interface SliderProps extends SliderCommonPropsAndState {
+	/**
+	 * An event emitted when slider values are changed
+	 *
+	 * Event payload equals to the updated slider values
+	 */
+	onValuesChange: (values: number[]) => void;
+}
+
+export interface SliderApi {}
+
+export interface SliderDirectives {
+	/**
+	 * Directive to get the slider component elementRef
+	 */
+	sliderDirective: Directive;
+
+	/**
+	 * Directive to get the minLabel elementRef
+	 */
+	minLabelDirective: Directive;
+
+	/**
+	 * Directive to get the maxLabel elementRef
+	 */
+	maxLabelDirective: Directive;
+}
+
+export interface SliderActions {
+	/**
+	 * Method to handle click on the slider
+	 * @param coordinate - new handle coordinate
+	 * @param handleNumber - id of the modified handle
+	 */
+	click(coordinate: number, handleNumber?: number): void;
+	/**
+	 * Method to process the keyboard event
+	 * @param event - keyboard event object
+	 * @param handleNumber - id of the modified handle
+	 */
+	onKeydown(event: KeyboardEvent, handleNumber: number): void;
+
+	/**
+	 * Method to manually invoke the slider size recomputation
+	 */
+	updateSliderSize(): void;
+
+	/**
+	 * Method describing the behavior of the slider handle on mouse down event
+	 * @param event - mouse event
+	 * @param handleId - numeric id of the handle
+	 */
+	onMouseDown(event: MouseEvent, handleId: number): void;
+}
```

Comment:
You should remove the "on" prefix (`mouseDown`, `keydown`) to make actions consistent (with `click` and with other widgets).